src/quartz_mailer.cr

```diff
@@ -6,21 +6,39 @@ class Quartz::Mailer
   def self.deliver(message : Message)
     config = Quartz.config
 
-    if config.smtp_enabled
-      EMail::Sender.new(
-        config.smtp_address, config.smtp_port,
-        use_tls: config.use_ssl,
-        logger: config.logger
-      ).start do
-        enqueue message.to_email
-      end
+    unless config.smtp_enabled
+      config.logger.warn "SMTP Disabled, not actually sending email."
+      return
+    end
+
+    if config.use_authentication
+      connect_and_send(
+        message,
+        config.smtp_address,
+        config.smtp_port,
+        use_tls: config.use_tls,
+        logger: config.logger,
+        auth: { config.username, config.password },
+      )
     else
-      puts "SMTP Disabled, not actually sending email"
+      connect_and_send(
+        message,
+        config.smtp_address,
+        config.smtp_port,
+        use_tls: config.use_tls,
+        logger: config.logger
+      )
     end
 
-  rescue e : EMail::Error::MessageError
-    puts "Email message couldn't be delivered."
-    puts e.message
+  end
+
+  private def self.connect_and_send(message : Message, smtp_address : String, smtp_port : Int32, **connection_options)
```

Comment:
Is this going to connect/disconnect for every email delivered?

Reply:
It does, yes (no change from before). And I discovered today that the connect happens synchronously even if the send is async. I started thinking about adding a Queue and spinning up the connection at boot time, but I think I need to learn more about Crystal fibers. 

I welcome any input on that, if you have thoughts.